ext/pcre/php_pcre.c

```diff
@@ -545,14 +554,21 @@ static zend_string **make_subpats_table(uint32_t num_subpats, pcre_cache_entry *
 		return NULL;
 	}
 
-	subpat_names = ecalloc(num_subpats, sizeof(zend_string *));
+	subpat_names = pecalloc(num_subpats, sizeof(zend_string *), persistent);
 	while (ni++ < name_cnt) {
 		unsigned short name_idx = 0x100 * (unsigned char)name_table[0] + (unsigned char)name_table[1];
 		const char *name = name_table + 2;
-		subpat_names[name_idx] = zend_string_init(name, strlen(name), 0);
+		/* Note: if we're making persistent strings, they will only be used within this thread.
+		 * Although we will be storing them in user-exposed arrays, they cannot cause problems
+		 * because they only live in this thread and the last reference is deleted on shutdown
+		 * instead of by user code. */
+		subpat_names[name_idx] = zend_string_init(name, strlen(name), persistent);
+		if (persistent) {
+			GC_MAKE_PERSISTENT_LOCAL(subpat_names[name_idx]);
+		}
 		if (is_numeric_string(ZSTR_VAL(subpat_names[name_idx]), ZSTR_LEN(subpat_names[name_idx]), NULL, NULL, 0) > 0) {
```

Comment:
Complete aside, but this could use `is_numeric_str_function` as a future refactoring

Reply:
I tried to cover this branch in a test, but I'm not so sure we can get here. Compilation seems to fail with a similar error before we get here it seems.

Reply:
I've removed this code (and redundant test I added) as per my previous comment.

Reply:
Possibly merge those commits already and rebase?